src/mlpack/methods/ann/layer/mean_pooling.hpp

```diff
@@ -209,10 +198,9 @@ class MeanPoolingType : public Layer<InputType, OutputType>
     // prefix sums performed (col-wise and row-wise).
     // We can use this to determine which method to use.
     const bool condition = (error.n_elem * kernelHeight * kernelWidth) >
-        (4 * error.n_elem + 2 * input.n_elem);
+        (4 * error.n_elem + 2 * output.n_elem);
```

Comment:
Are you sure this should be changed?  I don't follow what was wrong with the previous implementation.

Reply:
Nothing is wrong here. I just took output as we aren't really using the input variable other than for determining the size, and input & output size is same.

Reply:
Ok, I see now, thanks for the clarification.  But can you update the comment also?  It still refers to the input.  It's important to keep comments up to date, otherwise there will certainly be confusion later. :)

Reply:
Idts there is a need to update comments as the input area is equal to the output area.
Should I write at the top that the dimension of input and output is the same?